cpp/src/io/orc/stripe_data.cu

```diff
@@ -451,29 +453,18 @@ inline __device__ int decode_base128_varint(volatile orc_bytestream_s* bs, int p
 /**
  * @brief Decodes a signed int128 encoded as base-128 varint (used for decimals)
  */
-inline __device__ int128_s decode_varint128(volatile orc_bytestream_s* bs, int pos)
+inline __device__ __int128_t decode_varint128(volatile orc_bytestream_s* bs, int pos)
 {
-  uint32_t b        = bytestream_readbyte(bs, pos++);
-  int64_t sign_mask = -(int32_t)(b & 1);
-  uint64_t v        = (b >> 1) & 0x3f;
-  uint32_t bitpos   = 6;
-  uint64_t lo       = v;
-  uint64_t hi       = 0;
+  uint32_t b           = bytestream_readbyte(bs, pos++);
+  __int128_t sign_mask = -(int32_t)(b & 1);
+  __int128_t v         = (b >> 1) & 0x3f;
+  uint32_t bitpos      = 6;
   while (b > 0x7f && bitpos < 128) {
     b = bytestream_readbyte(bs, pos++);
     v |= ((uint64_t)(b & 0x7f)) << (bitpos & 0x3f);
```

Comment:
This could the the reason why numbers larger than int64::max get truncated.
```suggestion
    v |= ((__uint128_t)(b & 0x7f)) << bitpos;
```

Reply:
I made that change locally but it didn't seem to fix it.

Reply:
```
[----------] 12 tests from OrcWriterTest
[ RUN      ] OrcWriterTest.MultiColumn
/home/cph/gitlab/cudf/cpp/tests/utilities/column_utilities.cu:511: Failure
Failed
first difference: lhs[0] = 730460942017056408000, rhs[0] = 140165131658350756320
```

Reply:
Strange, when I applied this patch locally, the test started passing. Will test further with different values.